### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by Luchy-B**
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
- Do not use the bullet gem in this exercise - instead analyze the logs as showed in this article

**Before:** 
Post db4

Post db 4b

**After**
After

- User index page
- user show page:
- User post index page:
- post show page:

### CODE REVIEW

File: spec/integration_view/users/show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      visit user_path(user.id)
+      page.has_button?('See all posts')
+    end
+
+    it "When I click to see all posts, it redirects me to the user's post's index page." do
+      visit user_path(user.id)
+      click_on 'See all posts'
+      expect(page).to have_current_path("/users/#{user.id}/posts")
+    end
+  end
```
Comment (rloterh) [inline spec/integration_view/users/show_spec.rb]: - [ ] A test to assert that the scenario **_When I click a user's post, it redirects me to that post's show page_** for _user show view_ spec is missing! Please ensure to improve upon this aspect.

File: spec/integration_view/Post/index.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      expect(page).to have_content(@first_post.text)
+    end
+
+    it 'should show comments counter' do
+      expect(page).to have_content(@first_post.comments_counter)
+    end
+
+    it 'should show likes counter' do
+      expect(page).to have_content(@first_post.likes_counter)
+    end
```
Comment (rloterh) [inline spec/integration_view/Post/index.html.erb_spec.rb]: - [ ] Let's also ensure to include tests for the below scenarios for _user index_ views: 
   * I can see a section for pagination if there are more posts than fit on the view
   * When I click on a post, it redirects me to that post's show page

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Davian, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there!
🏁 🚦 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Nice job improving on the code!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
Let's improve the look of the ReadMe by removing the underlined:

image


Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — APPROVED** [review]
Hi @Luchy-B @imranhossainemi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Nice job improving on the code!
ReadMe looks great!

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Luchy-B** [comment]
Thank You for the Review

**Pull request merged**